cypress/e2e/permit2.test.ts

```diff
@@ -5,212 +5,193 @@ import { getTestSelector } from '../utils'
   function expectTokenAllowanceForPermit2ToBeMax() {
     // check token approval
-    return cy
-      .hardhat()
+    cy.hardhat()
       .then(({ approval, wallet }) => approval.getTokenAllowanceForPermit2({ owner: wallet, token: INPUT_TOKEN }))
       .should('deep.equal', MaxUint256)
   }
 
   /** Asserts the universal router has a max permit2 approval for spend of the input token on-chain. */
-  function expectPermit2AllowanceForUniversalRouterToBeMax(approvalTime: number) {
-    return cy
-      .hardhat()
+  function expectPermit2AllowanceForUniversalRouterToBeMax() {
+    cy.hardhat()
       .then((hardhat) => hardhat.approval.getPermit2Allowance({ owner: hardhat.wallet, token: INPUT_TOKEN }))
       .then((allowance) => {
         cy.wrap(MaxUint160.eq(allowance.amount)).should('eq', true)
         // Asserts that the on-chain expiration is in 30 days, within a tolerance of 40 seconds.
-        const expected = Math.floor((approvalTime + 2_592_000_000) / 1000)
+        const THIRTY_DAYS_SECONDS = 2_592_000
+        const expected = Math.floor(Date.now() / 1000 + THIRTY_DAYS_SECONDS)
         cy.wrap(allowance.expiration).should('be.closeTo', expected, 40)
       })
   }
 
-  it('swaps when user has already approved token and permit2', () => {
-    cy.hardhat().then(({ approval, wallet }) => {
-      approval.setTokenAllowanceForPermit2({ owner: wallet, token: INPUT_TOKEN })
-      approval.setPermit2Allowance({ owner: wallet, token: INPUT_TOKEN })
-    })
-    initiateSwap()
-    cy.get(getTestSelector('confirmation-close-icon')).click()
-    // Verifies that there is a successful swap notification.
-    cy.contains('Swapped').should('exist')
-  })
+  describe('approval process (with intermediate screens)', () => {
+    // Turn off automine so that intermediate screens are available to assert on.
+    beforeEach(() => cy.hardhat({ automine: false }))
 
-  it('swaps after completing full permit2 approval process', () => {
-    cy.hardhat().then(({ provider }) => {
-      cy.spy(provider, 'send').as('permitApprovalSpy')
-    })
-    initiateSwap()
-    cy.contains('Enable spending limits for DAI on Uniswap').should('exist')
-    cy.contains('Approved').should('exist')
+    it('swaps after completing full permit2 approval process', () => {
+      initiateSwap()
 
-    cy.contains('Allow DAI to be used for swapping').should('exist')
-    cy.contains('Confirm Swap').should('exist')
+      // Verify token approval
+      cy.contains('Enable spending limits for DAI on Uniswap')
+      cy.wait('@eth_sendRawTransaction')
+      cy.hardhat().then((hardhat) => hardhat.mine())
+      cy.get(getTestSelector('popups')).contains('Approved')
+      expectTokenAllowanceForPermit2ToBeMax()
 
-    cy.then(() => {
-      const approvalTime = Date.now()
+      // Verify permit2 approval
+      cy.contains('Allow DAI to be used for swapping')
+      cy.wait('@eth_signTypedData_v4')
+      cy.contains('Confirm Swap').click()
```

Comment:
this will only appear if the quote price changed beyond a certain threshold

Reply:
it actually never appeared in any test runs i've seen so far. i know we're slowing down the test but i'm not sure we should assume it will appear

Reply:
This will appear if there was _not_ a price update, I think.
See https://github.com/Uniswap/interface/blob/e2dd78fd0e05e91c5eb286f754404bbd81e80905/src/components/swap/SwapModalFooter.tsx#L156-L204

Reply:
we're both wrong :(
this is the final review screen, which happens to share the same text